### PR TITLE
Improved Documentation

### PR DESCRIPTION
**Pull request by dougbrn**
In it's current iteration, the stistools documentation is lacking in terms of providing information to the users about the package, mainly regarding it's installation and usage. This was recently made known to the STIS team thanks to a series of helpcalls that expressed frustration on trying to utilize the package based on the available documentation. I've gone through and overhauled the documentation, adding:

- A More In-Depth Introduction: Expanded upon the existing introduction, added a statement on the IRAF-to-Python transition, and split up the content tree into Pipeline Software and Analysis Tools
-  An Installation Guide: Many of our users are/will be transitioning to Python from IRAF/PyRAF, an in-documentation installation guide ensures that users who are migrating because they need stistools will be met with step-by-step instructions for getting stistools setup.
- A Recommended CRDS setup guide: The Python CRDS package is the best way for a user to get reference files for use with the stistools data analysis tools, so I've added a section detailing how one would get that setup.

Additonally, I've gone through and fixed some issues with the current documentation:

- x2d and wavecal: The x2d and wavecal pipeline components were missing from the documentation, added them in.
- Pipeline Software Examples: None of the stistools tools had any examples rendered in the documentation, I found out that this is partly a mistake, as the sphinx docstrings were not placed at the top of the file (they were below the imports), so they were not being rendered on the html webpage. I fixed this issue for all of the pipeline software. Also, I removed the PyRAF examples, which would send conflicting messages with the deprication warnings that are occurring in the stis IRAF/PyRAF package.

### CODE REVIEW

File: doc/source/gettingstarted.rst [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    $ export CRDS_PATH="$HOME/crds_cache"
+    $ export CRDS_SERVER_URL="https://hst-crds.stsci.edu"
+    $ export oref="${CRDS_PATH}/references/hst/oref/"
+
+The above syntax define where your personal copies of CRDS reference files will be stored and the CRDS server that is used.
+Then the following command may be used to assign and obtain the best references files:
+
+.. code-block:: sh
+
+    $ crds bestrefs --update-bestrefs --sync-references=1 --files *.fits
```
Comment (SaOgaz) [inline doc/source/gettingstarted.rst]: You might want to clarify here that the bestrefs will run on files currently in your working directly (current example), or where-ever you specify with the last input.

File: doc/source/index.rst [hunk 2]
```diff
@@ -3,30 +3,69 @@
...
+    The information here reflect the *latest* software info and might not
+    be in-sync with the
+    `STIS Data Handbook <http://www.stsci.edu/hst/stis/documents/handbooks/currentDHB/>`_.
+
+Relationship to the IRAF/PyRAF stis Package
+*******************************************
+
+The intent of this package is to provide the user community with many of the useful tools contained within the original IRAF/PyRAF
+stis package in Python. This is consistent with a larger effort to transition the user community from IRAF/PyRAF to Python,
+which can be read about `here <https://stak-notebooks.readthedocs.io/en/latest/index.html>`_. At this time, there remains several tools
```
Comment (SaOgaz) [inline doc/source/index.rst]: When this goes in, let's open an issue to add a better/additional link here once I'm able to post a knowledge base article about the end of IRAF support.

**Comment by sean-lockwood** [comment]
I'll take a look early next week.

File: doc/source/gettingstarted.rst [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+First, we need to configure Conda to grant it access to the AstroConda channel, which can be done by running the following command in a
+BASH shell.
+
+.. code-block:: sh
+
+    $ conda config --add channels http://ssb.stsci.edu/astroconda
+    # Writes changes to ~/.condarc
+
+Now we're ready to install the STScI Software Stack, we'll accomplish this by setting up a fresh Conda environment.
+Run the following command, you can change "astroconda" to be whatever name for the environment you wish.
```
[the remark this answers is not in the document]
Comment (dougbrn) [reply doc/source/gettingstarted.rst]: @SaOgaz this seems like a good change, I've incorporated it in the PR.

File: doc/source/gettingstarted.rst (hunk 1 above)
Comment (dougbrn) [reply doc/source/gettingstarted.rst]: Added a note at the end clarifying this.

**Pull request closed without merging**